crates/sui-storage/src/lock_service.rs

```diff
@@ -196,9 +309,18 @@ impl LockServiceImpl {
             }
         }
 
-        self.transaction_lock
-            .multi_insert(objects.iter().map(|obj_ref| (obj_ref, None)))?;
+        let write_batch = write_batch.insert_batch(
+            &self.transaction_lock,
+            objects.iter().map(|obj_ref| (obj_ref, None)),
+        )?;
+
+        Ok(write_batch)
+    }
 
+    fn initialize_locks(&self, objects: &[ObjectRef], is_force_reset: bool) -> SuiResult {
+        let write_batch = self.transaction_lock.batch();
+        let write_batch = self.initialize_locks_impl(write_batch, objects, is_force_reset)?;
```

Comment:
I don't think we should abstract the initialize_locks_impl.   There's no need to pay the cost of a batch for the simple initialize_locks case, only for the sequence case.